drukmakori_sivatag/src/test/java/JUnit/PipeTest.java

```diff
@@ -0,0 +1,160 @@
+package JUnit;
+
+import drukmakori_sivatag.*;
+import org.junit.Test;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+
+import static org.junit.Assert.*;
+
+public class PipeTest {
+
+    private Pipe pipe;
+
+    @Test
+    public void TestStepLeaking() {
+
+        pipe = new Pipe();
+        pipe.setBroken(true);
+
+        var src = new Source();
+        src.AddNeighbour(pipe);
+        src.Step();
+
+        pipe.Step();
+
+        assertEquals("Pipe should leak water", 1, pipe.getGc().getLeakedWater());
+    }
+
+    @Test
+    public void TestPipeFix() {
+        pipe = new Pipe();
+        pipe.setBroken(true);
+
+        pipe.Fix();
+
+        assertFalse("Pipe shouldn't be broken", pipe.isBroken());
+
+        assertTrue("TicksUntilBreakable should be between 0 and 10", (pipe.getTicksUntilBreakable() >= 0 && pipe.getTicksUntilBreakable() <= 9));
+    }
+
+    @Test
+    public void TestPipeAcceptFree() {
+        Mechanic mechanic = new Mechanic();
+        pipe = new Pipe();
+
+        pipe.Accept(mechanic);
+
+        assertTrue("Mechanic should occupy pipe", pipe.GetOccupied());
+    }
+
+    @Test
+    public void TestPipeAcceptOccupied() {
+        Mechanic mechanic1 = new Mechanic();
+        Mechanic mechanic2 = new Mechanic();
+        pipe = new Pipe();
+
+        var result1 = pipe.Accept(mechanic1);
+        assertTrue("Mechanic1 should occupy pipe", result1);
+
+        var result2 = pipe.Accept(mechanic2);
+        assertFalse("Pipe mustn't accept mechanic2", result2);
+    }
+
+    @Test
+    public void TestPipeAcceptSlippery() {
+        Mechanic mechanic = new Mechanic();
+        pipe = new Pipe();
+        ArrayList<Field> pumpNeighbors = new ArrayList<>();
+        pumpNeighbors.add(pipe);
+        Pump pump1 = new Pump();
+        ArrayList<Field> pipeNeighbors = new ArrayList<>();
+        pipeNeighbors.add(pump1);
+        pump1.SetNeighbors(pumpNeighbors);
+        pipe.SetNeighbors(pipeNeighbors);
+        mechanic.SetPosition(pump1);
+
+        pipe.SetSlippery(true);
+        var result1 = pipe.Accept(mechanic);
+        System.out.println("TestPipeAcceptSlippery 50% of failing must be solved");
```

Comment:
Azért van mert a  Pipe osztály Accept függvényében random sorsoljuk ki, hogy melyik szomszédra kerülünk, még akkor is, ha null az egyik szomszéd. Kérlek javítsátok ezt.

Reply:
Okés kijavítom!